### PR TITLE
Use external launcher to set Files as default explorer

### LINKED ISSUE #7574: Some Environment Variables are not accessible via Win + R when Files is set as the default file manager.
### Description

%APPDATA%,
%LOCALAPPDATA% 

are not accessible via Win + R when Files is set as the default file manager.

### Steps To Reproduce

1. Set Files as default file manager
2. Launch run
3. Type %LOCALAPPDATA%  or %APPDATA%

### Expected behavior

Files should open these folders like %HOMEPATH%

### Files Version

Version: 2.1.0.0 OS Version: 10.0.22000.376

### Windows Version

Windows 11 21H2  22000.376

### Relevant Assets/Logs

image
image

### LINKED ISSUE #8344: File crashes when I open the control panel
### Description

When i open a Files window and than control pannel, Files crash.

### Steps To Reproduce

1. Open a Files window
2. Open control pannel 
4. Files crash

### Expected behavior

Files should not close after opening contol pannel 

### Files Version

2.1.13

### Windows Version

Windows 11 Insider Preview 22543.1000 (rs_prerelease)

### Relevant Assets/Logs

[debug.log](https://github.com/files-community/Files/files/8057078/debug.log)
[debug_fulltrust.log](https://github.com/files-community/Files/files/8057079/debug_fulltrust.log)

https://user-images.githubusercontent.com/45063794/153783057-34b0ac6e-4227-43ab-a47a-1db4eb244b3e.mp4

### LINKED ISSUE #8626: Opening "shell:AppsFolder" from Run has odd behavior...
### Description

Whenever I open the "shell:AppsFolder" from run (Win+R) It opens then closes, then opens my documents folder in the old file explorer app (Despite me having the default registry thing enabled"

### Steps To Reproduce

1. Sideload the latest version
2. Turn on "Set files as the default file manager"
3. Press Win+R
4. Type in "shell:AppsFolder"
5. Press enter
6. Witness bug

### Expected behavior

It to open the Shell Apps folder in the Files app.

### Files Version

Version: 2.1.15.0 OS Version: 10.0.22000.556

### Windows Version

Edition	Windows 11 Home Version	21H2 Installed on	‎10/‎29/‎2021 OS build	22000.556 Experience	Windows Feature Experience Pack 1000.22000.556.0

### Relevant Assets/Logs

[debug.log](https://github.com/files-community/Files/files/8227854/debug.log)

### PR DESCRIPTION
**Pull request by gave92**
**Resolved / Related Issues**
- Fixes #8626
- Fixes #7574
- Fixes #8344 
- Implements (partially) #6382 

**Details of Changes**
Solves the issue of explorer not working when Files is uninstalled.
Adds some support for "show in folder".
Fixes the issue of Files window flashing briefly when opening control panel.
Fixes opening "%appdata%" from run window.

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

**Comment by yaichenbaum** [comment]
@gave92 is it connected to settings UI yet?

**Comment by gave92** [comment]
Yep

**Comment by yaichenbaum** [comment]
@gave92 is the launcher supposed to revert the changes when Files is uninstalled?

File: src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+	auto oleCleanup = wil::OleInitialize_failfast();
+
+	FILE* _debugStream = NULL;
+	PWSTR pszPath = NULL;
+	HRESULT hr = SHGetKnownFolderPath(FOLDERID_Desktop, 0, NULL, &pszPath);
+	if (SUCCEEDED(hr))
+	{
+		TCHAR debugPath[MAX_PATH];
+		wsprintf(debugPath, L"%s\\%s", pszPath, L"open_in_folder.txt");
+		//_wfreopen_s(&_debugStream, debugPath, L"w", stdout);
```
Comment (gave92) [inline src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp:36]: @yair100 try to uncomment this line, deploying and switching off-on the toggle. It should create a debug log on desktop. Could you post its contents after you try to invoke explorer with Files uninstalled?

Comment (gave92) [reply src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp:32]: Also please verify that this file does no longer exist when Files is uninstalled (you may have multiple release of Files) -> %LOCALAPPDATA%\\Microsoft\\WindowsApps\\files.exe

Comment (yaichenbaum) [reply src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp:32]: > Also please verify that this file does no longer exist when Files is uninstalled (you may have multiple releases of Files) -> %LOCALAPPDATA%\Microsoft\WindowsApps\files.exe

The file is removed on uninstall.

> What does it happen when you try and the call is not forwarded to explorer? You get the usual error message? Does it even work for you when Files is installed?

I get an error "Application not found".

Comment (yair100) [reply src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp:36]: > @yair100 try to uncomment this line, deploying and switching off-on the toggle. It should create a debug log on desktop. Could you post its contents after you try to invoke explorer with Files uninstalled?

It's not creating a debug file, is it posible the launcher isn't properly deplying?

Comment (gave92) [reply src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp:32]: The launcher should be copied to %LocalAppData%\Files\FilesLaucher.exe, does it exist? Does Files launch when it's installed?

Comment (yaichenbaum) [reply src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp:32]: The launcher is being copied and it's creating the log file now but nothing is being entered into it.

Comment (gave92) [reply src/Files.OpenDialog/FilesLauncher/FilesLauncher.cpp:32]: [f5a10e1](https://github.com/files-community/Files/pull/8684/commits/f5a10e1e858e739ba7d165789f15cc25fc867a64) should fix the issue.

**Comment by gave92** [comment]
Ouch it _does_ require admin / UAC prompt to uninstall.

**Review by yaichenbaum — APPROVED** [review]
LGTM

**Pull request merged**